data/src/main/scala/ch.epfl.scala.index.data/bintray/DownloadPoms.scala

```diff
@@ -1,90 +1,130 @@
+    val computed = md.digest(toVerify.getBytes("UTF-8")).map("%02x".format(_)).mkString
 
-  private def download(search: BintraySearch) = {
-    import search._
-    def escape(v: String) = v.replace(" ", "%20")
-    val downloadUri =
-      if(repo == "jcenter" && owner == "bintray") Uri(escape(s"https://jcenter.bintray.com/$path"))
-      else Uri(escape(s"https://dl.bintray.com/$owner/$repo/$path"))
-    HttpRequest(uri = downloadUri)
+    computed == sha1
   }
-  
-  val jcenterBintrayHttpFlow = Http().cachedHostConnectionPoolHttps[BintraySearch]("jcenter.bintray.com")
-  val dlBintrayHttpFlow      = Http().cachedHostConnectionPoolHttps[BintraySearch]("dl.bintray.com")
-  val splitConnectionPool =
-    Flow.fromGraph(GraphDSL.create() { implicit b =>
-      import GraphDSL.Implicits._
-      val broadcast = b.add(Broadcast[(HttpRequest, BintraySearch)](2))
-      val merge     = b.add(Merge[(Try[HttpResponse], BintraySearch)](2))
-
-      def hostIs(host: String)(in: (HttpRequest, BintraySearch)) = {
-        val (request, _) = in
-        request.uri.authority.host.toString == host
-      }
-      broadcast.out(0).filter(hostIs("jcenter.bintray.com")).via(jcenterBintrayHttpFlow) ~> merge.in(0)
-      broadcast.out(1).filter(hostIs("dl.bintray.com"))     .via(dlBintrayHttpFlow)      ~> merge.in(1)
-      FlowShape(broadcast.in, merge.out)
-    })
-
-  private val source = scala.io.Source.fromFile(bintrayCheckpoint.toFile)
-  private val searchesBySha1 = BintrayMeta.sortedByCreated(bintrayCheckpoint).
-    filter(s => !Files.exists(pomPath(s))). // todo make sure file content matches sha1!
-    groupBy(_.sha1).                        // remove duplicates with sha1
-    map{case (_, vs) => vs.head}
-
-  private val progress = new ProgressBar("Downloading POMs", searchesBySha1.size)
-  
-  private def pomPath(search: BintraySearch) = bintrayPomBase.resolve(s"${search.sha1}.pom")
-  
-  private val downloadPoms = Source(searchesBySha1)
-    .map(s => (download(s), s))
-    .via(splitConnectionPool)
-    .mapAsync(1){
-      case (Success(response), search) => {
-        Unmarshal(response).to[String].map(body =>
-          if(response.status == StatusCodes.OK) Right((body, search))
-          else Left((body, search))
-        )
+
+  /**
+   * verify the sha1 file hash
+   * @param path the path to the file to verify
+   * @param sha1 the sha1 hash
+   * @return
+   */
+  private def verifySHA1FileHash(path: Path, sha1: String): Boolean = {
+
+    val source = scala.io.Source.fromFile(path.toFile)
+    val content = source.mkString.split(nl).mkString("")
+    source.close()
+
+    verifyChecksum(content, sha1)
+  }
+  /**
+   * get a list of bintraySearch object where no sha1 file exists
+   */
+  private val searchesBySha1: Set[BintraySearch] = {
+
+    BintrayMeta.readQueriedPoms(bintrayCheckpoint)
+      .filter(s => !Files.exists(pomPath(s)) || !verifySHA1FileHash(pomPath(s), s.sha1))
```

Comment:
so per discussion it's because we pretty print
https://github.com/scalacenter/scaladex/blob/master/data/src/main/scala/ch.epfl.scala.index.data/bintray/DownloadPoms.scala#L77-L79